src/app/static/src/app/store/surveyAndProgram/mutations.ts

```diff
@@ -73,5 +77,13 @@ export const mutations: MutationTree<SurveyAndProgramState> = {
 
     [SurveyAndProgramMutation.Ready](state: ReadyState) {
         state.ready = true;
+    },
+
+    [SurveyAndProgramMutation.WarningsFetched](state: SurveyAndProgramState, action: PayloadWithType<Warning[]>){
+        state.warnings = [...state.warnings, ...action.payload];
```

Comment:
This isn't quite going to work - if the user doesn't manually clear the warnings for a type, but then re-uploads or pulls a file, the warnings for the previous upload will remain behind and just get the new ones appended (so won't be cleared if the warnings are fixed) - unlike, say, in the modelRun's `WarningsFetched` where re-running causes the previous run's warnings to be overwritten. 

So unfortunately, I think you're going to have to keep the warnings per SAP type separately in the state, and pass the type as a parameter to the mutation (or have three separate mutations) so it can overwrite the correct set of warnings,